clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/model/AmazonReservationReportBuilder.groovy

```diff
@@ -0,0 +1,229 @@
+
+  /**
+   * This version (v3) of the reservation report:
+   *
+   * - normalizes all regional reservations to `xlarge` instance types
+   * - attempts to cover all shortfalls with regional reservations (converting `xlarge` to the necessary instance types)
+   */
+  class V3 implements AmazonReservationReportBuilder {
+    // relative multipliers for instance types < xlarge where the baseline is xlarge
+    // (would be nice if AWS had an API to derive this from)
+    Map<String, Double> instanceTypeMultipliers = [
+      "t2.micro" : 0.25,
+      "t2.small" : 0.25,
+      "t2.medium": 0.5,
+      "t2.large" : 0.5,
+
+      "m1.small" : 0.25,
+      "m1.medium": 0.5,
+      "m1.large" : 0.5,
+      "m3.medium": 0.5,
+      "m3.large" : 0.5,
+      "m4.large" : 0.5,
+
+      "c1.medium": 0.25,
+      "c3.large" : 0.5,
+      "c4.large" : 0.5,
+
+      "r3.large" : 0.5,
+      "r4.large" : 0.5,
+
+      "i3.large" : 0.5
+    ]
+
+    AmazonReservationReport build(AmazonReservationReport source) {
+      def reservations = source.reservations.sort(
+        false, new AmazonReservationReport.DescendingOverallReservationDetailComparator()
+      )
+
+      // aggregate all regional reservations for each instance family
+      def regionalReservations = aggregateRegionalReservations(reservations)
+
+      // remove any regional reservations that have been fully utilized (ie. they've all been converted to xlarge)
+      reservations.removeAll { it.availabilityZone == "*" && it.totalSurplus() == 0 }
+
+      // add the aggregated regional reservations
+      reservations.addAll(regionalReservations)
+
+      // used to track the order in which regional reservations are used to cover shortfalls
+      def borrowIndex = new AtomicInteger(0)
+
+      def shortfalls = reservations.findAll { it.totalSurplus() < 0 }
+      shortfalls.each { OverallReservationDetail shortfall ->
+        def regional = regionalReservations.find {
+          it.instanceFamily() == shortfall.instanceFamily() && shortfall.region == it.region && it.os == shortfall.os
+        }
+
+        if (!regional) {
+          // no regional reservation to cover shortfall from
+          return
+        }
+
+        coverShortfall(borrowIndex, regional, shortfall)
+      }
+
+      return new AmazonReservationReport(
+        start: source.start,
+        end: source.end,
+        accounts: source.accounts,
+        reservations: reservations,
+        errorsByRegion: source.errorsByRegion
+      )
+    }
+
+    private List<OverallReservationDetail> aggregateRegionalReservations(List<OverallReservationDetail> reservations) {
+      def regionalReservations = filterRegionalReservations(reservations)
+
+      regionalReservations.groupBy { "${it.region}-${it.instanceFamily()}-${it.os}" }.collect {
+        def overallRegionalReservationDetail = new OverallReservationDetail(
+          availabilityZone: "*",
+          instanceType: it.value[0].instanceFamily() + ".xlarge",
```

Comment:
I need to double back on why we chose `Flarge`. I seem to recall there being a good reason, but I can't remember it.